internal/util-interface/src/main/java/xsbti/WorkspaceEdit.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package xsbti;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * A minimal representatin of the `WorkspaceEdit` found in the LSP protocol.
+ *
+ * <p>However it only supports the minimal `changes` to ensure the fixes will work with all clients.
+ *
+ * <p>NOTE: In the future this may be expanded to handle resource operations via `documentChanges`.
+ *
+ * @see <a href=
+ *     "https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceEdit">`WorkspaceEdit`</a>
+ */
+public interface WorkspaceEdit {
+
+  /** Map of the `TextEdit`s that belong to a specific URI. */
+  Map<URI, TextEdit> changes();
```

Comment:
We should avoid `Map` since this could introduce non-determinism when persisted into JSON, and `Problem` is a good candidate for caching.

Reply:
Sure, would you recommend something else specifically?

Reply:
`java.util.List<xsbti.T2<URI, TextEdit>>`, maybe?

Reply:
Huh, I really dislike the `get1()` and `get2()` that would need to be used for this. I instead created a `FileChanges` interface  that will hold a `URI` and the `List<TextEdits>` that belong to it. Originally this map was actually wrong as it should have been `Map<URI, List<TextEdit>>` so I'm glad we caught this 😬. But let me know if this new approach is alright.